File: Python/p012.py
#!/usr/bin/python3

from timeit import default_timer
from projecteuler import count_divisors

def main():
    start = default_timer()

    i = 0
    triang = 0
    finished = 0

    while not finished:
        i = i + 1
        triang = triang + i

        if count_divisors(triang) > 500:
            finished = 1

    end = default_timer()

    print('Project Euler, Problem 12')
    print('Answer: {}'.format(triang))

    print('Elapsed time: {:.9f} seconds'.format(end - start))

if __name__ == '__main__':
    main()

File: C/p012.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include "projecteuler.h"

int main(int argc, char **argv)
{
   int i = 0, finished = 0, count, triang = 0;
   double elapsed;
   struct timespec start, end;

   clock_gettime(CLOCK_MONOTONIC, &start);

   while(!finished)
   {
      i++;
      triang += i;
      count = count_divisors(triang);
      
      if(count > 500)
      {
         finished = 1;
      }
   }

   clock_gettime(CLOCK_MONOTONIC, &end);

   elapsed = (end.tv_sec - start.tv_sec) + (double)(end.tv_nsec - start.tv_nsec) / 1000000000;

   printf("Project Euler, Problem 12\n");   
   printf("Answer: %d\n", triang);

   printf("Elapsed time: %.9lf seconds\n", elapsed);

   return 0;
}
